### PR TITLE
Add moe.emmaexe.ntfyDesktop

### PR DESCRIPTION
**Pull request by emmaexe**
### Please confirm your submission meets all the criteria

- [X] Please describe the application briefly.
  This application is a desktop client for [ntfy](https://github.com/binwiederhier/ntfy). Its main purpose is to sit in the background, receive notifications from an ntfy server and show them on the desktop. It runs in the tray and has a GUI for configuring notification sources.

- [X] The domain used for the application ID is [controlled by the application developers][appid-domain] and the [application id guidelines][appid] are followed.
- [X] I have read the and followed all the [Submission and licence requirements][reqs].
- [X] I have [built][build] and tested the submission locally.
- [X] I am an author/developer/upstream contributor of the project. If not, I contacted upstream developers about this submission. **Link: https://github.com/emmaexe/ntfyDesktop**

[appid-domain]: https://docs.flathub.org/docs/for-app-authors/requirements/#control-over-domain-or-repository
[appid]: https://docs.flathub.org/docs/for-app-authors/requirements#application-id
[reqs]: https://docs.flathub.org/docs/for-app-authors/requirements
[build]: https://docs.flathub.org/docs/for-app-authors/submission/#before-submission

### CODE REVIEW

File: moe.emmaexe.ntfyDesktop.json [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+            ],
+            "sources": [
+                {
+                    "type": "archive",
+                    "url": "https://github.com/emmaexe/ntfyDesktop/archive/refs/tags/v1.2.0.tar.gz",
+                    "sha256": ""
+                }
+            ],
+            "modules": [
+                "ntfyDesktop-modules.json"
```
Comment (hfiguiere) [inline moe.emmaexe.ntfyDesktop.json:32]: this should be inline in this manifest.

File: moe.emmaexe.ntfyDesktop.json [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+            "buildsystem": "cmake-ninja",
+            "config-opts": [
+                "-DCMAKE_BUILD_TYPE=Release",
+                "-DND_BUILD_TYPE=Flatpak"
+            ],
+            "sources": [
+                {
+                    "type": "archive",
+                    "url": "https://github.com/emmaexe/ntfyDesktop/archive/refs/tags/v1.2.0.tar.gz",
+                    "sha256": ""
```
Comment (hfiguiere) [inline moe.emmaexe.ntfyDesktop.json:28]: sha25 is required.

File: flathub.json [hunk 3]
```diff
@@ -0,0 +1,3 @@
+{
+    "only-arches": ["x86_64"]
```
Comment (hfiguiere) [inline flathub.json:2]: why?

File: ntfyDesktop-modules.json [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+                    }
+                }
+            ]
+        },
+        {
+            "name": "emojicpp",
+            "buildsystem": "cmake-ninja",
+            "config-opts": [
+                "-DCMAKE_INSTALL_PREFIX=/app"
+            ],
```
Comment (hfiguiere) [inline ntfyDesktop-modules.json:47]:
```suggestion
```

this is not necessary.

File: moe.emmaexe.ntfyDesktop.json (hunk 2 above)
Comment (emmaexe) [reply moe.emmaexe.ntfyDesktop.json:28]: I will add it later once v1.2.0 releases.

File: flathub.json (hunk 3 above)
Comment (emmaexe) [reply flathub.json:2]: Why is this wrong?

Comment (hfiguiere) [reply flathub.json:2]: why only support this architecture?

Comment (emmaexe) [reply flathub.json:2]: I could build for both but I can only test for x86_64. In hindsight, I don't really see any reason why the app shouldn't work on arm. I could remove this.

**Comment by emmaexe** [comment]
bot, build moe.emmaexe.ntfyDesktop

File: ntfyDesktop-modules.json
Comment (bbhtt) [inline ntfyDesktop-modules.json:1]: this should be in the main manifest

**Comment by bbhtt** [comment]
Going to test some changes... please hold

**Pull request closed without merging**